internal/platform/select.go

```diff
@@ -0,0 +1,7 @@
+package platform
+
+import "time"
+
+func Select(n int, r, w, e *FdSet, timeout *time.Duration) (int, error) {
```

Comment:
I haven't discussed this change, the one adaptation I did for the platform API is to use a Duration instead of a Timeval. There is a caveat though: the `select` syscall actually mutates the pointed timeval value (e.g. if the `select(2)` call terminates early; e.g. NetBSD, cf.: https://github.com/golang/sys/blob/master/unix/syscall_unix_test.go#L606-L617

Reply:
maybe move your comments here to godoc?